Tests/NIOTests/DatagramChannelTests.swift

```diff
@@ -662,4 +666,91 @@ final class DatagramChannelTests: XCTestCase {
             XCTAssertFalse(try channel2.getOption(ChannelOptions.explicitCongestionNotification).wait())
         } ())
     }
+    
+    private func testEcnReceive(address: String, vectorRead: Bool, vectorSend: Bool) {
+        XCTAssertNoThrow(try {
+            let receiveBootstrap: DatagramBootstrap
+            if vectorRead {
+                receiveBootstrap = DatagramBootstrap(group: group)
+                    .channelOption(ChannelOptions.datagramVectorReadMessageCount, value: 4)
+            } else {
+                receiveBootstrap = DatagramBootstrap(group: group)
+            }
+                
+            let receiveChannel = try receiveBootstrap
+                .channelOption(ChannelOptions.explicitCongestionNotification, value: true)
+                .channelInitializer { channel in
+                    channel.pipeline.addHandler(DatagramReadRecorder<ByteBuffer>(), name: "ByteReadRecorder")
+                }
+                .bind(host: address, port: 0)
+                .wait()
+            defer {
+                XCTAssertNoThrow(try receiveChannel.close().wait())
+            }
+            let sendChannel = try DatagramBootstrap(group: group)
+                .bind(host: address, port: 0)
+                .wait()
+            defer {
+                XCTAssertNoThrow(try! sendChannel.close().wait())
```

Comment:
No point having an `XCTAssertNoThrow` if we crash ;)

```suggestion
                XCTAssertNoThrow(try sendChannel.close().wait())
```